app/uk/gov/hmrc/soletraderidentificationfrontend/services/SubmissionService.scala

```diff
@@ -25,61 +25,163 @@ import javax.inject.{Inject, Singleton}
             _ <- soleTraderIdentificationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
-          } yield
-            SoleTraderDetailsMismatch(failureReason)
+          } yield {
+            JourneyCompleted(journeyConfig.continueUrl)
+          }
       }
+
+    case Right(_) if individualDetails.optNino.isEmpty && isEnabled(EnableNoNinoJourney) => for {
+      _ <- createTrnService.createTrn(journeyId)
+      _ <- soleTraderIdentificationService.storeBusinessVerificationStatus(journeyId, BusinessVerificationUnchallenged)
+      _ <- soleTraderIdentificationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
     } yield {
-      response
+      JourneyCompleted(journeyConfig.continueUrl)
+    }
+
+    case Right(_) => for {
+      _ <- soleTraderIdentificationService.storeBusinessVerificationStatus(journeyId, BusinessVerificationUnchallenged)
+      _ <- soleTraderIdentificationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
+    } yield {
+      JourneyCompleted(journeyConfig.continueUrl)
     }
+
+    case Left(failureReason) =>
+      for {
+        _ <- soleTraderIdentificationService.storeBusinessVerificationStatus(journeyId, BusinessVerificationUnchallenged)
+        _ <- soleTraderIdentificationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
+      } yield
+        SoleTraderDetailsMismatch(failureReason)
+  }
+
+  private def handleIndividualJourney(journeyId: String,
+                                      matchingResult: Either[SoleTraderDetailsMatching.SoleTraderDetailsMatchFailure, Boolean],
+                                      continueUrl: String)
+                                     (implicit hc: HeaderCarrier,
+                                      ec: ExecutionContext): Future[SubmissionResponse] = matchingResult match {
+    case Right(true | false) =>
+      Future.successful(JourneyCompleted(continueUrl))
+
+    case Left(failureReason) =>
+      for {
+        _ <- soleTraderIdentificationService.storeBusinessVerificationStatus(journeyId, BusinessVerificationUnchallenged)
+        _ <- soleTraderIdentificationService.storeRegistrationStatus(journeyId, RegistrationNotCalled)
+      } yield
+        SoleTraderDetailsMismatch(failureReason)
+  }
+
+
+  private def calculateMatchingResult(journeyId: String, journeyConfig: JourneyConfig, individualDetails: IndividualDetails)
+                                     (implicit hc: HeaderCarrier, ec: ExecutionContext)
+  : Future[Either[SoleTraderDetailsMatching.SoleTraderDetailsMatchFailure, Boolean]] = {
+    if (individualDetails.optNino.isEmpty && isEnabled(EnableNoNinoJourney))
+      soleTraderMatchingService.matchSoleTraderDetailsNoNino(journeyId, individualDetails)
+    else soleTraderMatchingService.matchSoleTraderDetails(journeyId, individualDetails, journeyConfig)
+  }
+
+  sealed trait BusinessScenario
```

Comment:
I would put these definitions in a model class similar to BusinessVerificationStatus and RegistrationStatus.

Reply:
.... Richard I think that at the moment it is only used in this SubmissionService and makes non sense in another file .... can be seen as an implementation detail and in the new version I made even private to clarify it